HTMLSerializer.js

```diff
@@ -347,6 +413,21 @@ var HTMLSerializer = class {
   }
 
   /**
+   * Create a function that will generate unique strings which can be used as
+   * ids.
+   *
+   * @return {Function} A funtion that generates a unique string each time it is
+   *     called.
+   */
+  generateIDGenerator() {
```

Comment:
I just pushed a fix where I check if the id already exists, and added a test covering that case.